File: examples/go/quickstart.go
// Quickstart example for the SuperSandbox Go SDK.
//
// Run:
//
//	SUPERSANDBOX_API_KEY=<key> go run quickstart.go
package main

import (
	"context"
	"fmt"
	"os"
	"time"

	supersandbox "github.com/superagi/supersandbox-sdk/go/supersandbox"
)

func main() {
	apiKey := os.Getenv("SUPERSANDBOX_API_KEY")
	if apiKey == "" {
		fmt.Fprintln(os.Stderr, "SUPERSANDBOX_API_KEY is not set")
		os.Exit(1)
	}

	client := supersandbox.New(apiKey)
	ctx := context.Background()

	// ── Create ────────────────────────────────────────────────────────────────
	fmt.Println("Creating sandbox...")
	sb, err := client.Create(ctx, supersandbox.CreateSandboxParams{
		Image:          supersandbox.ImageSpec{URI: "python:3.11-slim"},
		Entrypoint:     []string{"sleep", "120"},
		ResourceLimits: supersandbox.ResourceLimits{"cpu": "250m", "memory": "256Mi"},
		Metadata:       map[string]string{"example": "quickstart"},
	})
	if err != nil {
		fmt.Fprintf(os.Stderr, "create: %v\n", err)
		os.Exit(1)
	}
	fmt.Printf("  %s\n", sb)

	defer func() {
		fmt.Println("Deleting sandbox...")
		_ = sb.Delete(context.Background())
		fmt.Println("  Done.")
	}()

	// ── Run a task ────────────────────────────────────────────────────────────
	fmt.Println("Running task...")
	task, err := sb.Tasks().Submit(ctx, `python3 -c "print('hello from sandbox')"`)
	if err != nil {
		fmt.Fprintf(os.Stderr, "submit: %v\n", err)
		os.Exit(1)
	}
	fmt.Printf("  task id=%s\n", task.EffectiveID())

	// Poll until done
	taskID := task.EffectiveID()
	for i := 0; i < 30; i++ {
		time.Sleep(time.Second)
		task, err = sb.Tasks().Get(ctx, taskID)
		if err != nil {
			fmt.Fprintf(os.Stderr, "get task: %v\n", err)
			os.Exit(1)
		}
		if task.Status != nil && (*task.Status == "completed" || *task.Status == "failed") {
			break
		}
	}
	if task.Status != nil {
		fmt.Printf("  status=%s exit_code=%v\n", *task.Status, task.ExitCode)
	}

	// Fetch logs
	result, err := sb.Tasks().Logs(ctx, taskID, nil)
	if err != nil {
		fmt.Fprintf(os.Stderr, "logs: %v\n", err)
		os.Exit(1)
	}
	fmt.Printf("  output: %q\n", result.Logs)

	// ── Container logs ────────────────────────────────────────────────────────
	fmt.Println("Fetching container logs...")
	logs, err := sb.GetLogs(ctx, 20)
	if err != nil {
		fmt.Fprintf(os.Stderr, "get logs: %v\n", err)
		os.Exit(1)
	}
	fmt.Printf("  %q\n", logs[:min(len(logs), 120)])
}

func min(a, b int) int {
	if a < b {
		return a
	}
	return b
}

File: examples/python/quickstart.py
"""Quickstart — create a sandbox, run a command, delete it."""

import asyncio
import os
from supersandbox import AsyncSuperSandbox

API_KEY = os.environ["SUPERSANDBOX_API_KEY"]


async def main() -> None:
    async with AsyncSuperSandbox(api_key=API_KEY) as client:
        print("Creating sandbox...")
        sb = await client.create(
            image="python:3.11-slim",
            entrypoint=["sleep", "300"],
            resource_limits={"cpu": "500m", "memory": "512Mi"},
            metadata={"example": "quickstart"},
        )
        print(f"  {sb}")

        print("Running task...")
        task = await sb.tasks.submit("python3 -c \"print('hello from sandbox')\"")
        print(f"  task id={task.id} status={task.status}")

        # Poll until done
        import asyncio as _a
        for _ in range(10):
            task = await sb.tasks.get(task.id)
            if task.status in ("completed", "failed"):
                break
            await _a.sleep(1)

        result = await sb.tasks.logs(task.id)
        print(f"  output: {result.logs.strip()!r}")
        print(f"  exit code: {task.exit_code}")

        print("Fetching logs...")
        logs = await sb.get_logs(tail=10)
        print(f"  {logs.strip()!r}")

        print("Deleting sandbox...")
        await sb.delete()
        print("Done.")


if __name__ == "__main__":
    asyncio.run(main())
